### PR TITLE
Adds tests script for deployed hma

### PR DESCRIPTION
**Pull request by BarrettOlson**
Summary
---------

This is an attempt to add a smoke test to our hma development flow by adding a Make target `dev_test_instance` This target will pass the new root level `output.tf` values added in this PR to a new python script `tests/scripts/check_deployed_instance.py`

For the script to work you will need to have successfully deployed (via `terraform apply` or `make dev_create_instance` AND have non-expired AWS credentials locally.   

(This PR also copies b.jpg from the pdq directory to be used in the test) 

I spent too much time trying to be smart with Make and realized that is probably beyond me. Open to suggestions on the flow. Various parts of this will need to change as we improve/change pieces of HMA but the methods should be mostly reusable.  

Additional:
I changed the 'SK' for hash sent to the db back to 'type#pdq' from 'type:pdq'

Test Plan
---------

```
$ make dev_test_instance
TF_DYNAMODB_TABLE="barrett-HMADataStore" \
        TF_BUCKET_NAME="barrett-hashing-data20210302200326365300000003" \
        TF_PREFIX="barrett" \
        ./tests/utils/check_deployed.py

Checking status of deployed instance...
Running against instance using prefix: barrett
Attempting to upload threat_exchange_data/pdq.te...
Looking for index/pdq_hashes.index in barrett-hashing-data20210302200326365300000003 (check every 5s x20)...
Attempting to upload images/test_photo.jpg...
Looking for {'PK': 'c#images/test_photo.jpg', 'SK': 'type#pdq'} in dynamodb.Table(name='barrett-HMADataStore') (check every 5s x20)...
Looking for {'PK': 'c#images/test_photo.jpg', 'SK': 'te#5555555555555555'} in dynamodb.Table(name='barrett-HMADataStore') (check every 5s x20)...
Cleaning up...   
Deleted s3 objects and datastore items

Success! Deployed instance seems to behave as expected!
Remember to run "terraform destroy" or "make dev_destroy_instance" if you are done with your deployed instance.
```

### CODE REVIEW

File: hasher-matcher-actioner/terraform/outputs.tf
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/outputs.tf:3]: ...seplling

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (jeberl) [inline hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:80]: Why do we do this in a loop. Why might it not be there first time?

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (jeberl) [inline hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:99]: New to python testing. Is this standard to have 1 function that runs all the tests

**Review by jeberl — COMMENTED** [review]
This seems right to me. The empty file is supposed to be there right?

Going to wait for one other person to review before accepting

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (BarrettOlson) [reply hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:80]: The lambda needs time to do its job and for things to propagate as both uploads trigger an event the call the lambdas

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (BarrettOlson) [reply hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:99]: Not for a pytest tests but this script is basically just running a demo of the system so I think it makes sense here... could be convinced otherwise.

File: hasher-matcher-actioner/hmalib/dto.py
Comment (schatten) [inline hasher-matcher-actioner/hmalib/dto.py:27]: 🙏🏽

File: hasher-matcher-actioner/terraform/outputs.tf
Comment (schatten) [inline hasher-matcher-actioner/terraform/outputs.tf:5]: Can you describe the bucket? Would love it if the variable name gave an idea of what I'm supposed to do with the bucket.

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (schatten) [inline hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:108]: If I am testing an instance the second time, this object will already exist. In those cases, can we check that the index exists and proceed? 

Or perhaps you think that is not an end-to-end test. What do you think?

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (schatten) [inline hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:115]: This also assumes the images/ S3 prefix-directory-thingy is empty.

For this test to be repeatable, I'd add some randomness to the file name.

File: hasher-matcher-actioner/tests/scripts/check_deployed_instance.py
Comment (schatten) [inline hasher-matcher-actioner/tests/scripts/check_deployed_instance.py:131]: Non-blocking. If you can find the time and the answers, can you document the dynamodb table design somewhere? Having it handy would help.

**Review by schatten — APPROVED** [review]
This is already in pretty good shape. I have one nit in the inline comments. But you can address that. Approving to unblock.

Feedback
---
I don't find the design of the check_deployed_instance.py file "super" open to re-use. There are methods, but they are generic names, so hard to understand what they would do. 

eg. What does "run()" do? If I were to re-use this method, what would happen....

Also, I could create a BaseTest module which maintains a set of `s3_resource_keys` and `dynamodb_item_pks` and provides methods to clean up after you are done. The smoke test and other integration tests would extend this class and benefit from being focused on what they do rather than do the boilerplaty things.

Similarly, each integration test can be thought of as being an list of sub-flows. Each sub flow is (action, wait-for-something, assert). But it is too early to tell.

However, and this is important
---
I feel when we are writing the second test (integration or smoke), we should be able to refactor as required. Right now, I completely understand that it is too early to tell what the "right" abstractions are.

File: hasher-matcher-actioner/terraform/outputs.tf
Comment (BarrettOlson) [reply hasher-matcher-actioner/terraform/outputs.tf:5]: Followed up offline, keeping names generic for now as we use the same bucket for various different purposes. (someday it might make sense to have 'data_upload_bucket' and 'index_store_bucket' for example).

**Review by BarrettOlson — COMMENTED** [review]
Thanks @schatten and @jeberl! add some inline comments and agree we should break it up more in the future.

**Pull request merged**
